spec/models/journey_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Journey, type: :model do
+  it { is_expected.to belong_to(:move) }
+  it { is_expected.to belong_to(:supplier) }
+  it { is_expected.to belong_to(:from_location) }
+  it { is_expected.to belong_to(:to_location) }
+  it { is_expected.to validate_presence_of(:move) }
+  it { is_expected.to validate_presence_of(:supplier) }
+  it { is_expected.to validate_presence_of(:from_location) }
+  it { is_expected.to validate_presence_of(:to_location) }
+  it { is_expected.to validate_presence_of(:client_timestamp) }
```

Comment:
what do we mean with  `client`?

Reply:
client usually means the supplier; but it could also be a police station or PMU, or indeed any thing which could trigger events on a move